attestation/verification/verification.go

```diff
@@ -0,0 +1,209 @@
+package verification
+
```

Comment:
A suggestion would be to use generics to accommodate other types in the future. 

TBH, I don't know every bit and I could be totally wrong. 

``` go
const (
	Name = "policyverify"
	Type = slsa.VerificationSummaryPredicate
)

var (
	_ attestation.Subjecter = &Attestor[cryptoutil.Verifier]{}
	_ attestation.Attestor  = &Attestor[cryptoutil.Verifier]{}
)

type Verifier interface {
	Verify(body io.Reader, signature []byte) error
}

type Attestor[T Verifier] struct {
	slsa.VerificationSummary

	policyEnvelope   dsse.Envelope
	policyVerifiers  []T
	collectionSource source.Sourcer
	subjectDigests   []string
}

type Option[T Verifier] func(*Attestor[T])

func VerifyWithPolicyVerifiers[T Verifier](policyVerifiers []T) Option[T] {
	return func(vo *Attestor[T]) {
		vo.policyVerifiers = append(vo.policyVerifiers, policyVerifiers...)
	}
}

func VerifyWithSubjectDigests[T Verifier](subjectDigests []cryptoutil.DigestSet) Option[T] {
	return func(vo *Attestor[T]) {
		for _, set := range subjectDigests {
			for _, digest := range set {
				vo.subjectDigests = append(vo.subjectDigests, digest)
			}
		}
	}
}

func VerifyWithCollectionSource[T Verifier](source source.Sourcer) Option[T] {
	return func(vo *Attestor[T]) {
		vo.collectionSource = source
	}
}

func New[T Verifier](opts ...Option[T]) *Attestor[T] {
	a := &Attestor[T]{}
	for _, opt := range opts {
		opt(a)
	}

	return a
}

func (vs *Attestor[T]) Name() string {
	return Name
}

func (vs *Attestor[T]) Type() string {
	return Type
}

func (vs *Attestor[T]) RunType() attestation.RunType {
	return attestation.ExecuteRunType
}

func (vs *Attestor[T]) Subjects() map[string]cryptoutil.DigestSet {
	panic("not implemented") // TODO: Implement
}
```

Reply:
Thanks for the suggestion. I don't think we have a current use case to support some other form of verifier outside of a `cryptoutil.Verifier` here currently, and we currently need our verifiers to implement the Verifier interface.